src/analysis/processing/qgsalgorithmextractlabels.cpp

```diff
@@ -0,0 +1,461 @@
+}
+
+QStringList QgsExtractLabelsAlgorithm::tags() const
+{
+  return QObject::tr( "map themes,font,position" ).split( ',' );
+}
+
+QgsProcessingAlgorithm::Flags QgsExtractLabelsAlgorithm::flags() const
+{
+  return QgsProcessingAlgorithm::flags() | FlagRequiresProject;
+}
+
+QString QgsExtractLabelsAlgorithm::group() const
+{
+  return QObject::tr( "Cartography" );
+}
+
+QString QgsExtractLabelsAlgorithm::groupId() const
+{
+  return QStringLiteral( "cartography" );
+}
+
+void QgsExtractLabelsAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterExtent(
+                  QStringLiteral( "EXTENT" ),
+                  QObject::tr( "Map extent" ) ) );
+
+  addParameter( new QgsProcessingParameterScale(
+                  QStringLiteral( "SCALE" ),
+                  QObject::tr( "Map scale" ) ) );
+
+  std::unique_ptr<QgsProcessingParameterMapTheme> mapThemeParameter = std::make_unique<QgsProcessingParameterMapTheme>(
+        QStringLiteral( "MAP_THEME" ),
+        QObject::tr( "Map theme" ),
+        QVariant(), true );
+  mapThemeParameter->setHelp( QObject::tr( "This parameter is optional. When left unset, the algorithm will fallback to match the currently opened project's layer tree visibility to extract labels from." ) );
+  addParameter( mapThemeParameter.release() );
+
+  addParameter( new QgsProcessingParameterBoolean(
+                  QStringLiteral( "INCLUDE_UNPLACED" ),
+                  QObject::tr( "Include unplaced labels" ),
+                  QVariant( true ), true ) );
+
+  std::unique_ptr<QgsProcessingParameterNumber> dpiParameter = std::make_unique<QgsProcessingParameterNumber>(
+        QStringLiteral( "DPI" ),
+        QObject::tr( "Map resolution (in DPI)" ),
+        QgsProcessingParameterNumber::Double,
+        QVariant( 96.0 ), true );
+  dpiParameter->setFlags( dpiParameter->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( dpiParameter.release() );
+
+  addParameter( new QgsProcessingParameterFeatureSink(
+                  QStringLiteral( "OUTPUT" ),
+                  QObject::tr( "Extracted labels" ),
+                  QgsProcessing::TypeVectorPoint ) );
+}
+
+QString QgsExtractLabelsAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Converts map labels to a point layer with relevant details saved as attributes." );
+}
+
+QString QgsExtractLabelsAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm extracts label information from a rendered map at a given extent and scale.\n\n"
+                      "If a map theme is provided, the rendered map will match the visibility and symbology of that theme. If left blank, the project's layer tree setup will be used.\n\n"
+                      "Extracted label information include their position (served as point geometries), the associated layer name and feature ID, as well as label text, rotation, alignment, and font details." );
```

Comment:
```suggestion
                      "Extracted label information include\n"
                      " - position (served as point geometries)\n"
                      " - the associated layer name and feature ID\n"
                      " - label text\n"
                      " - rotation in degree, clockwise\n"
                      " - alignment\n"
                      " - font details" );
```

Can you check the rotation (cw/ccw) and adjust alignment according to my other comment where required?

Reply:
@m-kuhn , rotation details added (it was actually missing from the default style attached to the output layer, good catch), and description tweaked. I haven't used a bullet list, that doesn't play well with the formatting we apply to short help strings.

Reply:
We should support md here :-)

Reply:
@m-kuhn , yeah, it'd be nice (would likely output OK as ASCII for qgis_process too)